src/Microsoft.AspNetCore.Sockets.Client.Http/HttpConnection.cs

```diff
@@ -166,13 +176,43 @@ private Task StartAsyncCore()
                         return;
                     }
 
-                    _transport = _transportFactory.CreateTransport(GetAvailableServerTransports(negotiationResponse));
-                    connectUrl = CreateConnectUrl(Url, negotiationResponse);
-                }
+                    // This should only need to happen once
+                    _serverTransports = GetAvailableServerTransports(negotiationResponse);
+                    connectUrl = CreateConnectUrl(Url, negotiationResponse.ConnectionId);
+
+                    foreach (var transport in AllTransports)
+                    {
+                        try
+                        {
+                            if ((transport & _serverTransports & _requestedTransportType) != 0)
+                            {
+                                // The negotiation response gets cleared in the fallback scenario.
+                                if (negotiationResponse == null)
+                                {
+                                    negotiationResponse = await GetNegotiationResponse();
+                                    connectUrl = CreateConnectUrl(Url, negotiationResponse.ConnectionId);
+                                }
 
-                Log.StartingTransport(_logger, _transport, connectUrl);
-                await StartTransport(connectUrl);
+                                Log.StartingTransport(_logger, transport, connectUrl);
+                                await StartTransport(connectUrl, transport);
+                                break;
+                            }
+                        }
+                        catch (Exception)
+                        {
+                            // Try the next transport
+                            // Clear the negotiation response so we know to re-negotiate.
+                            negotiationResponse = null;
+                        }
+                    }
+                }
+                
+                if (_transport == null)
+                {
+                    throw new InvalidOperationException("No transport was created.");
```

Comment:
Let's finesse this a bit: "Unable to connect to the server with any of the available transports".